src/main/java/io/aiven/kafka/connect/common/output/parquet/ParquetSchemaBuilder.java

```diff
@@ -90,7 +102,27 @@ private Schema headersSchema(final SinkRecord record) {
         return SchemaBuilder.map().values(avroData.fromConnectSchema(headerSchema));
     }
 
-    private Schema avroSchemaFor(final SinkRecord record) {
+    protected Schema avroSchemaFor(final SinkRecord record) {
+        if (!envelopeEnabled) {
+            // envelope can be disabled only in case of single field
+            final OutputField field = fields.iterator().next();
+            final Schema schema = outputFieldSchema(field, record);
+            if (schema.getType() == Schema.Type.MAP) {
```

Comment:
this logic a bit wrong it will fail for others complex types like `Enums`, `Arrays` etc.
Docs: https://avro.apache.org/docs/current/spec.html#schema_complex